packages/vite-plugin/src/proxy-middleware.ts

```diff
@@ -0,0 +1,247 @@
+/**
+ * This modules provides the proxy located at
+ * /motion-canvas-proxy/...
+ *
+ * It is needed when acessing remote resources.
+ * Trying to access remote resources works while
+ * in preview, but will fail when you try to
+ * output the image (= "read" the canvas)
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image
+ * for reasons
+ *
+ * Using the proxy circumvents CORS-issues because
+ * this way all remote resources are served from the
+ * same server (localhost:9000) as the main app.
```

Comment:
nit: `localhost:9000` doesn't have to be mentioned here since it's not guaranteed to be the hostname, say "from the same domain as the main app" instead